src/game/editor/editor.cpp

```diff
@@ -3168,6 +3173,8 @@ void CEditor::RenderFileDialog()
 		UI()->DoLabel(&FileBoxLabel, "Search:", 10.0f, CUI::ALIGN_LEFT);
 		if(DoEditBox(&m_FilesSearchBoxID, &FileBox, m_aFileDialogFilterString, sizeof(m_aFileDialogFilterString), 10.0f, &m_FilesSearchBoxID))
 		{
+			// recreate newly filtered list
+			FilelistPopulate(m_FileDialogStorageType);
```

Comment:
This would do the whole file search again when typing a character. Would be better to work with the already gathered list here and update the needed information.

Reply:
Yes I know :/ But it worked smooth in my maps folder with 157 maps. Also I did not want to mess up the code too much by writing crappy cache code. Might look at ddnet how they handled it.